napi_js_props.md

```diff
@@ -0,0 +1,437 @@
+status = napi_create_string_utf8(env, "hello", -1, &value);
+if (status != napi_ok) return status;
+
+// arr[123] = 'hello';
+status = napi_set_element(env, arr, 123, value);
+if (status != napi_ok) return status;
+```
+
+You can also get properties using the APIs described in this section.
+Consider the following JavaScript snippet:
+```js
+var arr = [];
+return arr[123];
+```
+
+The following is the approximate equivalent of the N-API counterpart:
+```C
+napi_status status = napi_status_generic_failure;
+
+// var arr = []
+napi_value arr, value;
+status = napi_create_array(env, &arr);
+if (status != napi_ok) return status;
+
+// get arr[123]
+status = napi_get_element(env, arr, 123, &value);
+if (status != napi_ok) return status;
+```
+
+Finally, you can also define multiple properties on an object for performance
+reasons. Consider the following JavaScript:
+```js
+var obj = {};
+Object.defineProperties(obj, {
+  'foo': { value: 123, writable: true, configurable: true, enumerable: true },
+  'bar': { value: 456, writable: true, configurable: true, enumerable: true }
+});
+```
+
+The following is the approximate equivalent of the N-API counterpart:
+```C
+napi_status status = napi_status_generic_failure;
+
+// var obj = {};
+napi_value obj;
+status = napi_create_obj(env, &obj);
+if (status != napi_ok) return status;
+
+// Create napi_values for 123 and 456
+napi_value fooValue, barValue;
+status = napi_create_number(env, 123, &fooValue);
+if (status != napi_ok) return status;
+status = napi_create_number(env, 456, &barValue);
+if (status != napi_ok) return status;
+
+// Set the properties
+napi_property_descriptors descriptors[] = {
+  { "foo", fooValue, 0, 0, 0, napi_default, 0 },
+  { "bar", barValue, 0, 0, 0, napi_default, 0 }
+}
+status = napi_define_properties(env,
+                                obj,
+                                sizeof(descriptors) / sizeof(descriptors[0]),
+                                descriptors);
+if (status != napi_ok) return status;
+```
+
+## Structures
+
+### *napi_property_attributes*
+
+#### Definition
+```C
+typedef enum {
+  napi_default = 0,
+  napi_read_only = 1 << 0,
+  napi_dont_enum = 1 << 1,
+  napi_dont_delete = 1 << 2,
+  napi_static_property = 1 << 10,
+} napi_property_attributes;
+```
+
+### Description
+`napi_property_attributes` are flags used to control the behavior of properties 
+set on a JavaScript value. They roughly correspond to the attributes listed in
+[Section 6.1.7.1](https://tc39.github.io/ecma262/#table-2) of the
+[ECMAScript Language Specification](https://tc39.github.io/ecma262/). They can
+be one or more of the following bitflags:
+
+- `napi_default` - Used to indicate that no explicit attributes are set on the 
+given property. By default, a property is Writable, Enumerable, and
+ Configurable. Note that this is a deviation from the ECMAScript specification,
+ where generally the values for a property descriptor attribute default to 
+ false if they're not provided.
```

Comment:
This behavior along with the enum value names is going to change soon, based on discussion in https://github.com/nodejs/abi-stable-node/issues/221.